crates/win-api-wrappers/src/netmgmt.rs

```diff
@@ -60,74 +58,61 @@ pub fn get_local_admin_group_members() -> anyhow::Result<Vec<Sid>> {
 
 pub fn get_local_group_members(group_name: String) -> anyhow::Result<Vec<Sid>> {
     let group_name = WideString::from(group_name);
-    let group_members_ptr: *mut LOCALGROUP_MEMBERS_INFO_0 = std::ptr::null_mut();
-    let mut entries_read = 0;
+    let mut group_members: *mut u8 = std::ptr::null_mut();
+    let mut number_of_entries_read = 0;
     let mut total_entries = 0;
 
-    // SAFETY: All buffers are valid.
-    // WideString holds a null-terminated UTF-16 string, and as_pcwstr() returns a valid pointer to it.
-    // Specifying `MAX_PREFERRED_LENGTH` allocates the required amount of memory for the data, and
-    // the function will not return `ERROR_MORE_DATA`
-    unsafe {
-        let rc = NetLocalGroupGetMembers(
+    // SAFETY:
+    // - group_name holds a null-terminated UTF-16 string, and as_pcwstr() returns a valid pointer to it.
+    // - Specifying `MAX_PREFERRED_LENGTH` allocates the required amount of memory for the data, and the function will not return `ERROR_MORE_DATA`.
+    let ret = unsafe {
+        NetLocalGroupGetMembers(
             None,
             group_name.as_pcwstr(),
             0,
-            &mut group_members_ptr.cast(),
+            &mut group_members,
```

Comment:
Using `.cast()` or `as *mut _` to convert the pointer will create a temporary value that we then immediately borrow and _drop_. The original variable is left unchanged. We need work with the right type from the start, and convert _afterwards_.

Working example: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=de03e34c057c117ab745625c5acca955

Buggy example: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=7a61ab03938a721b8aeebf97bf0a7322